### PR TITLE
N°5793 - HTML Sanitizer: Allow 'start' attribute in 'ol' tag

### PR DESCRIPTION
**Pull request by jbostoen**
Just an example here. This fork is based on Combodo's Mail to Ticket Automation, which relies on internal methods.

The real contents look like this in MS Outlook (text censored):
image

The contents in iTop look like this:
image

HTML source:

```
<ol style="margin-top:0in" start="1" type="1"><li class="MsoListParagraph" style="margin-left:0in;mso-list:l0 level1 lfo1">xxx.</li></ol>

<p class="MsoNormal"> </p>

<ol style="margin-top:0in" start="2" type="1"><li class="MsoListParagraph" style="margin-left:0in;mso-list:l0 level1 lfo1">xxxx.</li></ol>
```

After stripping: there is indeed a paragraph in between, and iTop will ignore the "start".

### CODE REVIEW

**Review by Molkobain — APPROVED** [review]

**Comment by Molkobain** [comment]
Thanks for the PR Jeffrey! Makes total sense, will push it forward during next reviews.

**Review by Hipska — APPROVED** [review]

**Review by piRGoif — APPROVED** [review]
As a reference : https://www.w3.org/TR/html4/struct/lists.html#edef-OL

**Comment by Molkobain** [comment]
The idea was accepted during the technical review.
We would like you to update the corresponding unit test though, it can be found in test/core/sanitizer.

_PR will be reviewed functionally next week anyway to save time for the next iteration._

**Review by Molkobain — CHANGES_REQUESTED** [review]
Update corresponding unit test please.

**Comment by jbostoen** [comment]
Do I just need to edit utf-8_wrong_character_email_truncated.txt ?
Or create a new file? (and if so, do I need to add a reference to it somewhere in another file?)

**Comment by Molkobain** [comment]
I think it should rather be `whitelist_test.html` but its content doesn't seem right. @piRGoif ?

**Comment by piRGoif** [comment]
Hello !

About the test, I would add the new attribute in the \Combodo\iTop\Test\UnitTest\Core\Sanitizer\HTMLDOMSanitizerTest::WhiteListProvider method, to check if the attribute is kept after the sanitize run.

Maybe add also a new data set in \Combodo\iTop\Test\UnitTest\Core\Sanitizer\HTMLDOMSanitizerTest::RemoveBlackListedTagContentProvider, to check for OL with attributes to remove and the new one to keep ?

**Comment by Molkobain** [comment]
Accepted during functional review.
Will be merged once unit test is updated.

**Comment by jbostoen** [comment]
Would this qualify?

I extended the PR a bit based on info from W3Schools:
* https://www.w3schools.com/tags/tag_li.asp
* https://www.w3schools.com/tags/tag_ol.asp

Basically:
* li: also allow 'value' (to set a starting value)
* ol: also allow 'reversed' (decreasing list), 'type' (see link above) and 'start' (to set a starting value).

I quickly fiddled around, but didn't see the real difference between li > value and ol > start. ( https://stackoverflow.com/questions/33374212/difference-value-and-start-in-html-lists )

**Review by Molkobain — APPROVED** [review]

**Comment by Molkobain** [comment]
Yes, it's closely related :)

File: test/core/sanitizer/HTMLDOMSanitizerTest.php [hunk 1]
```diff
@@ -213,6 +213,12 @@ public function RemoveBlackListedTagContentProvider()
 				'html' => '<html><head></head><body lang="EN-GB" link="#0563C1" vlink="#954F72">bar</body></html>',
 				'expected' => 'bar',
 			),
+			
+			'list test' => array(
```
Comment (Molkobain) [inline test/core/sanitizer/HTMLDOMSanitizerTest.php]:
```suggestion
			'ordered list with attributes' => array(
```

**Comment by jbostoen** [comment]
Mind to also update https://www.itophub.io/wiki/page?id=latest:admin:rich_text_limitations

**Comment by Molkobain** [comment]
Done but will only appear once 3.1 wiki is published :)

**Pull request merged**